Marlin/src/HAL/HAL_DUE/usb/sd_mmc_spi_mem.cpp

```diff
@@ -2,9 +2,10 @@
  * Interface from Atmel USB MSD to Marlin SD card
  */
 
+#include "../../../inc/MarlinConfig.h"
+
 #if defined(ARDUINO_ARCH_SAM) && defined(SDSUPPORT)
```

Comment:
This should probably be like this as platform guards need to cover the entire source file,
``` cpp
   #ifdef ARDUINO_ARCH_SAM
   #include "../../../inc/MarlinConfig.h"
   #if ENABLED(SDSUPPORT)
```

Reply:
Yes and no. As we are dealing with .h files, it should be enough to guard the c/cpp files that include the .h file...
But in the sd_mmc_spi_mem.cpp, you are right. The include should be inside the guard... (even if i suspect it will make no difference at all , at least here)

Reply:
This is to help compilation speeds as much as avoid possible conflicts (I've already encounter some because of this in other platforms) as all Platform source files are always build. I would love to specify the appropriate platform by adding only that folder to the build, but we need to maintain Arduino IDE compatibility.

Reply:
@p3p : In fact, there is a dirty trick: We could rename all the *.c/*.cpp platform files to a different extension, say .plat, and then include them conditionally from a central .cpp file ;)

Reply:
oh god don't even say things like that, I think we will stick with the preprocessor guards ><